### PR TITLE
Aslr init

### PR DESCRIPTION
**Pull request by jenswi-linaro**
This should take care of the recent growth in the unpaged part of OP-TEE when compiled with paging enabled.

The memory usage diff is:
``` diff
%diff -u tee.mem_usage-orig tee.mem_usage
--- tee.mem_usage-orig	2020-02-26 13:26:38.761709263 +0100
+++ tee.mem_usage	2020-02-26 13:26:52.153834824 +0100
@@ -1,19 +1,21 @@
 RAM Usage        0E100000 - 0E168000 size 00068000 416 KiB 104 pages
-.text            0E100000 - 0E10E0C0 size 0000E0C0  56 KiB
-*hole*           0E10E0C0 - 0E10F000 size 00000F40   3 KiB
-.rodata          0E10F000 - 0E113380 size 00004380  16 KiB
-.gnu.hash        0E113380 - 0E113398 size 00000018   0 KiB
-.got             0E113398 - 0E1134FC size 00000164   0 KiB
-.ARM.exidx       0E1134FC - 0E115DBC size 000028C0  10 KiB
-.ARM.extab       0E115DBC - 0E1165FC size 00000840   2 KiB
-*hole*           0E1165FC - 0E117000 size 00000A04   2 KiB
-.data            0E117000 - 0E1185B0 size 000015B0   5 KiB
-.bss             0E1185B0 - 0E1197D8 size 00001228   4 KiB
-.heap1           0E1197D8 - 0E11C000 size 00002828  10 KiB
-.nozi            0E11C000 - 0E125B80 size 00009B80  38 KiB
-.heap2           0E125B80 - 0E134000 size 0000E480  57 KiB
-.rodata_init     0E134000 - 0E134378 size 00000378   0 KiB
-*hole*           0E134378 - 0E135000 size 00000C88   3 KiB
+.text            0E100000 - 0E108E30 size 00008E30  35 KiB
+*hole*           0E108E30 - 0E109000 size 000001D0   0 KiB
+.rodata          0E109000 - 0E10BFC0 size 00002FC0  11 KiB
+.gnu.hash        0E10BFC0 - 0E10BFD8 size 00000018   0 KiB
+.got             0E10BFD8 - 0E10C13C size 00000164   0 KiB
+.ARM.exidx       0E10C13C - 0E10EA0C size 000028D0  10 KiB
+.ARM.extab       0E10EA0C - 0E10F24C size 00000840   2 KiB
+*hole*           0E10F24C - 0E110000 size 00000DB4   3 KiB
+.data            0E110000 - 0E1115B0 size 000015B0   5 KiB
+.bss             0E1115B0 - 0E1127D8 size 00001228   4 KiB
+.heap1           0E1127D8 - 0E114000 size 00001828   6 KiB
+.nozi            0E114000 - 0E11DB80 size 00009B80  38 KiB
+.heap2           0E11DB80 - 0E12D000 size 0000F480  61 KiB
+.text_init       0E12D000 - 0E132290 size 00005290  20 KiB
+*hole*           0E132290 - 0E133000 size 00000D70   3 KiB
+.rodata_init     0E133000 - 0E134758 size 00001758   5 KiB
+*hole*           0E134758 - 0E135000 size 000008A8   2 KiB
 .rodata_pageable 0E135000 - 0E14383E size 0000E83E  58 KiB
 *hole*           0E14383E - 0E144000 size 000007C2   1 KiB
 .text_pageable   0E144000 - 0E168000 size 00024000 144 KiB
```

### CODE REVIEW

File: core/arch/arm/mm/core_mmu.c [hunk 1]
```diff
@@ -1204,8 +1204,11 @@ static struct tee_mmap_region *get_tmp_mmap(void)
  * based on the seed and return the offset from the link address.
  *
  * If an error happened: core_init_mmu_map is expected to panic.
+ *
+ * Note: this function is weak just to make it possible to exclue it from
```
Comment (jforissier) [inline core/arch/arm/mm/core_mmu.c]: exclude

File: core/arch/arm/kernel/generic_boot.c [hunk 2]
```diff
@@ -1208,9 +1208,13 @@ static void init_secondary_helper(unsigned long nsec_entry)
 	DMSG("Secondary CPU Switching to normal world boot");
 }
 
-void generic_boot_init_primary(unsigned long pageable_part,
-			       unsigned long nsec_entry __maybe_unused,
-			       unsigned long fdt)
+/*
+ * Note: this function is weak just to make it possible to exclue it from
```
Comment (jforissier) [inline core/arch/arm/kernel/generic_boot.c]: exclude

**Comment by jenswi-linaro** [comment]
Spell errors fixed.

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by jenswi-linaro** [comment]
Tag applied

**Pull request merged**